apache/datadog_checks/apache/apache.py

```diff
@@ -68,40 +70,40 @@ def check(self, instance):
         else:
             self.service_check(service_check_name, AgentCheck.OK, tags=service_check_tags)
         self.log.debug("apache check succeeded")
-        metric_count = 0
+        metric_found = False
         # Loop through and extract the numerical values
         for line in r.iter_lines(decode_unicode=True):
-            values = line.split(': ')
-            if len(values) == 2:  # match
-                metric, value = values
-                try:
-                    value = float(value)
-                except ValueError:
-                    continue
-
-                # Special case: kBytes => bytes
-                if metric == 'Total kBytes':
-                    value = value * 1024
-
-                # Send metric as a gauge, if applicable
-                if metric in self.GAUGES:
-                    metric_count += 1
-                    metric_name = self.GAUGES[metric]
-                    self.gauge(metric_name, value, tags=tags)
-
-                # Send metric as a rate, if applicable
-                if metric in self.RATES:
-                    metric_count += 1
-                    metric_name = self.RATES[metric]
-                    self.rate(metric_name, value, tags=tags)
-
-        if metric_count == 0:
-            if self.assumed_url.get(instance['apache_status_url'], None) is None and url[-5:] != '?auto':
-                self.assumed_url[instance['apache_status_url']] = '%s?auto' % url
+            try:
+                metric, value = line.split(': ')
+                value = float(value)
+            except ValueError:
+                continue
+
+            # Special case: kBytes => bytes
+            if metric == 'Total kBytes':
+                value = value * 1024
+
+            # Send metric as a gauge, if applicable
+            try:
+                self.gauge(self.GAUGES[metric], value, tags=tags)
```

Comment:
what about:

```
gauge_metric = self.GAUGES.get(metric, None)
if gauge_metric is not None:
    self.gauge(gauge_metric, value, tags=tags)
    metric_found = True
```
That way you still avoid the double lookup but you are not using exceptions for flow control

Reply:
```python

    setup = """
def do_something(g):
    pass
d = {'gauge': 5, 'other': 6}
"""
    print(timeit.timeit("""v = d.get('gauge', None)
if v is not None:
    do_something(v)""", setup=setup))
    print(timeit.timeit("""try:
    do_something(d['gauge'])
except KeyError:
    pass""", setup=setup))
```
0.1312337
0.0932472

It's faster with KeyError

Reply:
By the way, I have an idea to make it even nicer ;-)
I will submit something in that today.

Reply:
It's faster because it never raises `KeyError` in your case. Try with a case where it does and you'll see it's slower.

Reply:
Yes indeed, that's what i was saying "but if the metric is not in the dict it's not worth."
I will change that if you prefer the other method, it really depends on the usage, i am preparing something much better i hope any way, i will update the PR accordingly